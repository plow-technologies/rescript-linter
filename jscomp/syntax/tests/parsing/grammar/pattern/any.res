let _ = 1;
let _ | _ = 1;
let (_ : int) = 1;
let (_ | _ : int) = 1;
let ((_ | _) : int) = 1; // note: same ast as line above
let (_ :unit) | (_ : unit) = 1;
let _ as _x = 1;
let _ | _ as _x = 1;
let _ | (_ as _x) = 1;
let (_ | _) as _x = 1;
let _ as _y | _ as _x = 1;
let (_ as _y) | (_ as _x) = 1; // note: same ast as line above

switch () {
| _ => ()
| _ | _ => ()
| _ as _x => ()
| _ as _x | _ as _x => ()
| (_ : unit) => ()
| (_ : unit) | (_ : unit) => ()
}

let f = (_) => ()
let f = (_ as _x) => ()
let f = (_ : unit) => ()
let f = (_ : unit) => ()
let f = ((_ : unit) as _x) => ()

let g = (a, _) => ()
let g = (_, a) => ()

for _ in 0 to 10 { () }
for _ as _x in 0 to 10 { () }

for (_ in 0 to 10) { () }
for (_ as _x in 0 to 10) { () }

for ((_ : int) in 0 to 10) { () }
for ((_ : int) as _x in 0 to 10) { () }
