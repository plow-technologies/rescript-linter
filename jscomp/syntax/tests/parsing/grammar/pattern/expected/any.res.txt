let _ = 1
let _|_ = 1
let (_ : int) = 1
let ((_|_) : int) = 1
let ((_|_) : int) = 1
let (_ : unit)|(_ : unit) = 1
let _ as _x = 1
let _|_ as _x = 1
let _|_ as _x = 1
let _|_ as _x = 1
let _ as _y|_ as _x = 1
let _ as _y|_ as _x = 1
;;match () with
  | _ -> ()
  | _|_ -> ()
  | _ as _x -> ()
  | _ as _x|_ as _x -> ()
  | (_ : unit) -> ()
  | (_ : unit)|(_ : unit) -> ()
let f _ = ()
let f (_ as _x) = ()
let f (_ : unit) = ()
let f (_ : unit) = ()
let f ((_ : unit) as _x) = ()
let g a _ = ()
let g _ a = ()
;;for _ = 0 to 10 do () done
;;for _ as _x = 0 to 10 do () done
;;for _ = 0 to 10 do () done
;;for _ as _x = 0 to 10 do () done
;;for (_ : int) = 0 to 10 do () done
;;for (_ : int) as _x = 0 to 10 do () done